### DISCUSSION TITLE
Can leiden algorithm achieve hierarchical ?

### DISCUSSION THREAD

**Discussion by 7thParkk** [microsoft/graphrag#723] [Algorithm + Paper]
How is it layered when it comes to community detect? Where is it in the code? Looking forward to your reply

**Comment by natoverse** [comment]
We use the hierarchical Leiden implementation from graspologic described [here](https://graspologic-org.github.io/graspologic/latest/reference/reference/partition.html).

Our invocation is [here](https://github.com/microsoft/graphrag/blob/f5c9c2bee0065dab01e9563d396223db3407ed55/graphrag/index/verbs/graph/clustering/strategies/leiden.py#L51)
